tests/UNITTESTS/CurrentTime/test_CurrentTime.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2020, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "gtest/gtest.h"
+
+#include "ble/BLE.h"
+#include "ble/Gap.h"
+#include "ble/GattServer.h"
+#include "ble/GattClient.h"
+#include "ble/SecurityManager.h"
+
+#include "ble_mocks.h"
+
+/* This test does not test anything, you may use it as a template for your unit tests.
+ * It shows all the elements you need to use mocks for all the ble APIS. */
+
+class TestLinkLoss : public testing::Test {
+protected:
+    void SetUp()
+    {
+        ble = &BLE::Instance();
+    }
+
+    void TearDown()
+    {
+    }
+
+    BLE* ble;
+};
+
+TEST_F(TestLinkLoss, reset)
```

Comment:
That's an excellent explanation of the process and it could serve as a template for testing. 
Can you extend the fixture so it includes things that are common for BLE testing like the event queue the chainable event handlers ? 

I think the file can be moved some place else and naming of the fixture can reflect this is a template/example.

Reply:
chainable event handler is not a fake or mock, you use it same way, I don't think we need to duplicate its docs